aiomcache/pool.py

```diff
@@ -39,29 +37,20 @@ def acquire(self):
 
         :return: ``tuple`` (reader, writer)
         """
-        while self._size < self._minsize:
+        while self.size() == 0 or self.size() < self._minsize:
             _conn = yield from self._create_new_conn()
-
-            # Could not create new connection
             if _conn is None:
                 break
-            yield from self._pool.put(_conn)
+            self._pool.put_nowait(_conn)
```

Comment:
Maybe I'm missing something but I don't see it happening. `put_nowait` is sync. It calls `_put` that directly inserts into the internal deque:

```python
    def _put(self, item):
        self._queue.append(item)
```
This means that after `self._pool.put_nowait`, the object is inserted without breaking the coroutine execution.

Also, I added this tests that should cover this case (note it checks every time that the pool size is never bigger than maxsize): https://github.com/aio-libs/aiomcache/pull/46/files#diff-44251e44a09d028bd96e8275ea1a0ec9R92